cmd/client/main.go

```diff
@@ -238,15 +237,16 @@ func (c *Coordinator) doPoll(client *http.Client) error {
 	return nil
 }
 
-func (c *Coordinator) loop(bo backoff.BackOff, client *http.Client) {
-	op := func() error {
-		return c.doPoll(client)
+func (c *Coordinator) loop(ctx context.Context, bo backoff.BackOff, client *http.Client) {
+	op := func() (int, error) {
+		return 0, c.doPoll(client)
 	}
 
 	for {
-		if err := backoff.RetryNotify(op, bo, func(_ error, _ time.Duration) {
+		_, err := backoff.Retry(ctx, op, backoff.WithBackOff(bo), backoff.WithMaxElapsedTime(time.Duration(0)), backoff.WithNotify(func(_ error, _ time.Duration) {
```

Comment:
backoff.WithMaxElapsed(time.Duration(0)) doesn't appear to do much : https://github.com/cenkalti/backoff/blob/aa5bd1344128d6d985c9606fe384663cc506e1d8/retry.go#L122-L124

Reply:
Nvm : https://github.com/cenkalti/backoff/blob/aa5bd1344128d6d985c9606fe384663cc506e1d8/retry.go#L10